pkg/storage/v2/mysql/client.go

```diff
@@ -199,3 +207,29 @@ func (c *client) RunInTransaction(ctx context.Context, tx Transaction, f func()
 	}
 	return err
 }
+
+func (c *client) RunInTransactionV2(ctx *context.Context, f func(tx Transaction) error) error {
```

Comment:
We don't need to use pointer here, since `context.Context` is interface already, reference: https://github.com/bucketeer-io/bucketeer/blob/main/pkg/account/storage/v2/storage.go#L66-L73

Reply:
@Ubisoft-potato 
Thank you for review 👍 

In this PR, Transaction instances are managed using `Context.WithValue()`.
https://github.com/bucketeer-io/bucketeer/blob/update-storage-client-auto-ops/pkg/storage/v2/mysql/client.go#L212-L216

Also, each MySQLClient side uses `Context.Value` with `Client.Qe()` in between to check whether there is a Transaction instance and switch the instance to execute the query.

[Client.Qe()](https://github.com/bucketeer-io/bucketeer/blob/update-storage-client-auto-ops/pkg/storage/v2/mysql/client.go#L229-L235)

ex: `AutoOpsStorage.CreateAutoOpsRule()`
https://github.com/bucketeer-io/bucketeer/blob/update-storage-client-auto-ops/pkg/autoops/api/api.go#L195

https://github.com/bucketeer-io/bucketeer/blob/update-storage-client-auto-ops/pkg/autoops/storage/v2/auto_ops_rule.go#L71

I think that if no pointer is passed, the context will not be updated and the transaction instance will not be saved.
As a result, I think transactions won't work and each query will be executed separately.

What do you think?

Reply:
Instead of passing `tx` here, we can pass `ctx`:
```go
func (c *client) RunInTransactionV2(ctx context.Context, f func(ctx context.Context) error) error {

  ctx = context.WithValue(ctx, transactionKey, tx)

  if err = f(ctx); err == nil {
      err = tx.Commit()
  }

}

``` 

So we can still access the newset `tx` in the `ctx`.

Reply:
Hi @kakcy , I want to quote a small paragraph in https://www.oreilly.com/library/view/concurrency-in-go/9781491941294/ (it's a very good book)

<img width="400" alt="image" src="https://github.com/user-attachments/assets/e577f20f-b2c5-4d9d-bb5e-fc682d68604c" />

This book has a chapter about context package, and it notes that we should not pass by reference but using the instance instead. As the book also states, go philosophy is _"sharing by communicating, not communicating by sharing"_, so maybe we can find a way to not update the context `ctx` but using its tree-forked paradigm to share the transaction `tx` as @Ubisoft-potato stated? 

What do you think?

Reply:
Thank you for your detailed explanations.
I will modify it to match the philosophy of the Golang.